docs/migration/README.md

```diff
@@ -0,0 +1,106 @@
+
+### 3. Install Open Service Broker for Azure
+
+Follow the [doc](https://github.com/Azure/open-service-broker-azure/tree/master/contrib/cf) to install OSBA.
+
+### 4. Migrate service instances records from MASB database(the SQL one) to OSBA database(the redis one).
+
+1. Download migration scripts:
+
+```
+curl -L -O https://raw.githubusercontent.com/Azure/meta-azure-service-broker/master/docs/migration/migrate_to_new_azure_service_broker.sh
+curl -L -O https://raw.githubusercontent.com/Azure/meta-azure-service-broker/master/docs/migration/migrate_to_new_azure_service_broker.js
+curl -L -O https://raw.githubusercontent.com/Azure/meta-azure-service-broker/master/docs/migration/migrate_to_new_azure_service_broker_helper_pc.go
+curl -L -O https://raw.githubusercontent.com/Azure/meta-azure-service-broker/master/docs/migration/migrate_to_new_azure_service_broker_helper_bc.go
+```
+
+2. Get the `manifest.yml`(filled for `cf push`) of both MASB and OSBA.
+
+3. Run:
+
+```
+$ chmod +x migrate_to_new_azure_service_broker.sh
+$ ./migrate_to_new_azure_service_broker.sh <path-to-masb-manifest> <path-to-osba-manifest> (<path-to-new-sql-server-list>)
+```
+
+The broker database migration succeeds if the script ends with no failed instance IDs, bindings IDs, or other error messages. If not, please check the [troubleshooting](#troubleshooting) below.
+
+### 4. Update service broker via CF CLI
+
+Run:
+
+```
+$ cf update-service-broker <service-broker-name> <admin-to-new-broker> <password-to-new-broker> <url-to-new-broker>
+```
+
+The `<service-broker-name>` can be checked by `cf service-brokers` if you forget which name meta-azure-service-broker was registered with.
+
+## Notes (IMPORTANT)
+
+* The bound apps possiblely need some code changes, as the format of credentials delivered by the broker have some changes. Please refer to [open-service-broker-azure docs](https://github.com/Azure/open-service-broker-azure/tree/master/docs/modules) and [meta-azure-service-broker docs](https://github.com/Azure/meta-azure-service-broker/tree/master/docs) to see the differences module-by-module in detail. Note that the credentials just change in new bindings. Credentials in existing bindings won't change.
+
+* The provisioning parameters changes. If you use script to provide provisioning parameters for new service instance creation, remember to update it.
+
+## Troubleshooting
+
+* If you hit `invalid_client` issue, `tenant` issue, or `subscription` issue:
+
+  Please check following credentials in the MASB manifest:
+    * SUBSCRIPTION_ID
+    * TENANT_ID
+    * CLIENT_ID
+    * CLIENT_SECRET
+  
+* If you hit `Broken instance record` issue:
+  
+  Please confirm if it's an orphan record, the resouce on Azure in no longer existed, and the service instance is not in use. If yes, delete it in MASB database by running `delete from bindings where instanceID='<instanceID>'; delete from instances where instanceID='<instanceID>';` and retry the migration script.
```

Comment:
Maybe users need to know how to login the MASB database.

Reply:
I think there are too many tools to connect to a SQL DB. And the operators have the credentials. It should be very easy for them.